### PR TITLE
Change ts compilation target to es6 as default

### PR DESCRIPTION
**Pull request by vishal423**
Considering evergreen browsers support almost all ES6 features, we should enable it as default (reference: https://kangax.github.io/compat-table/es6/). It will reduce bundle size by few KiB.

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by deepu105** [comment]
Ya lets do it for v6

**Comment by pvliss** [comment]
Closing and re-opening to relaunch CI tests

**Comment by wmarques** [comment]
@vishal423  Looks like you have some issues on travis builds: https://travis-ci.org/jhipster/generator-jhipster/jobs/504932072

**Comment by vishal423** [comment]
@wmarques, those failures seems to be intermittent selector issues in protractor suite. Can you restart travis build to check again?

**Comment by vishal423** [comment]
Following Travis sample, I am able to reproduce build issues locally with production profile. I will debug further to see what can be done. Meantime, I have changed status to WIP

**Comment by vishal423** [comment]
I am almost done from my side. If build passes, then, this is good to review, test, & merge

**Comment by vishal423** [comment]
Travis and Azure build check passed. Finally able to nail down after few sleepless nights :sweat_smile:

File: generators/client/templates/angular/webpack/webpack.prod.js.ejs [hunk 1]
```diff
@@ -113,7 +113,10 @@ module.exports = webpackMerge(commonConfig({ env: ENV }), {
                 parallel: true,
                 cache: true,
                 terserOptions: {
+                    ecma: 6,
                     ie8: false,
+                    toplevel: true,
+                    module: true,
```
Comment (wmarques) [inline generators/client/templates/angular/webpack/webpack.prod.js.ejs]: why are we adding those two options ?

Comment (vishal423) [reply generators/client/templates/angular/webpack/webpack.prod.js.ejs]: Docs:

> module (default false) — Use when minifying an ES6 module. "use strict" is implied and names can be mangled on the top scope. If compress or mangle is enabled then the toplevel option will be enabled.
toplevel (default false) - set to true if you wish to enable top level variable and function name mangling and to drop unused variables and functions.

**Comment by wmarques** [comment]
@vishal423 thanks ! 😄

**Pull request merged**